tests/unit/grains/core_test.py

```diff
@@ -28,6 +28,15 @@
 
 # Globals
 core.__salt__ = {}
+core.__opts__ = {}
```

Comment:
_Update 2:_ Actually it is also a very good idea to Unit test what happens if `getaddrinfo` raises an exception. So I'd extend this test by mocking this method (as I wrote above) and try to mock it raises a `socket.error` exception, so the `log.warning` (what? why _a warning_, it is _an error_!) is there and nothing bad is happening further.